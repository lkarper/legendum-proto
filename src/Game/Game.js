import React from 'react';
import { Route } from 'react-router-dom';
import Story from '../Story/Story';
import Exercise from '../Exercise/Exercise';

const Game = (props) => {
    return (
        <div>
            <Route 
                path="/game/story/:chapt"
                component={Story}
            />
            <Route 
                path="/game/exercise/:chapt"
                component={Exercise}
            />
        </div>
    );
}

export default Game;