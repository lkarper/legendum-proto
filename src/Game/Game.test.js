import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Game from './Game';

describe('Game component', () => {
  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Game />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Game />
      </BrowserRouter>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
