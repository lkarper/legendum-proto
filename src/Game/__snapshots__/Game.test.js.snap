// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Game>
      <Route
        component={[Function]}
        path="/game/story/:chapt"
      />
      <Route
        component={[Function]}
        path="/game/exercises/:chapt"
      />
      <Route
        component={[Function]}
        path={
          Array [
            "/game/exercises/:chapt",
            "/game/story/:chapt",
          ]
        }
      />
    </Game>
  </Router>
</BrowserRouter>
`;
